core/src/client.rs

```diff
@@ -0,0 +1,16 @@
+//! Defines the `RpcClient` trait.
+use crate::{Error, IoHandler};
+use crate::futures::future::{self, FutureResult};
+
+/// The RpcClient trait. Use the #[rpc] macro derives the Request and Response
+/// generation.
+pub trait RpcClient: Send + Sync + 'static {
+    /// Calls an rpc method.
+    fn call_method(&self, request: String) -> FutureResult<String, Error>;
+}
+
+impl RpcClient for IoHandler {
+    fn call_method(&self, request: String) -> FutureResult<String, Error> {
+        future::ok(self.handle_request_sync(&request).unwrap())
```

Comment:
We can't be sure that `request` is not a `Notification` (no id), so the `unwrap` here seems invalid. Perhaps turn it into an error.